faiss_vector_posting.go

```diff
@@ -318,12 +325,41 @@ func (sb *SegmentBase) InterpretVectorIndex(field string, except *roaring.Bitmap
 					return rv, nil
 				}
 
-				scores, ids, err := vecIndex.SearchWithoutIDs(qVector, k, vectorIDsToExclude, params)
-				if err != nil {
-					return nil, err
+				var scores []float32
+				var ids []int64
+				var err error
+
+				if !requireFiltering {
+					scores, ids, err = vecIndex.SearchWithoutIDs(qVector, k,
+						vectorIDsToExclude, params)
+					if err != nil {
+						return nil, err
+					}
+				} else if eligibleDocIDs.Stats().Cardinality > 0 {
+					// Non-zero documents eligible per the filter query.
+
+					// Making doubly sure that the deleted docs do not make their
+					// way into the "inclusion" list, even though that's been
+					// accounted for when computing the eligible doc IDs.
+					eligibleDocIDs.AndNot(except)
+
+					// vector IDs corresponding to the local doc numbers to be
+					// considered for the search
+					vectorIDsToInclude := make([]int64, eligibleDocIDs.Stats().Cardinality)
```

Comment:
Lot of allocations here - which is not good ..
* `make` of an array of int64s
* `ToArray()` below allocates an array as well

Ideally just ToArray() should've sufficed, but if you absolutely cannot work with the return type, then you should just iterate over the bitmap to populate `vectorIDsToInclude` as opposed to converting it and then reconverting it - which is quite bad!

Reply:
VectorIDs are int64 hashes of the original float32 vector. Hence, the conversion.